### DISCUSSION TITLE
Request to add link

### DISCUSSION THREAD

**Discussion by JJBRT** [classgraph/classgraph#540] [Ideas]
Could you add [Burningwave Core](https://github.com/burningwave/core#burningwave-core-) to "[Alternatives](https://github.com/classgraph/classgraph#alternatives)" paragraph?

**Comment by lukehutch** [comment]
Done, no problem at all -- looks like a great project!

However in the name of friendly competition ( 😃 ), I'm surprised at the claim on the main page that it is "THE MOST POWERFUL CLASSPATH SCANNER"... can it do all of this?:

https://github.com/classgraph/classgraph/wiki/Classpath-Specification-Mechanisms

or all of this?:

https://github.com/classgraph/classgraph/wiki/Code-examples

ClassGraph is also highly multithreaded, and also uses `DirectByteBuffer` instances where needed for speed (after very careful profiling)... and it even includes a highly optimized custom zipfile central directory parser so that it can scan jars within jars, without extracting the inner jar first. (The Java `ZipFile` API is locked to be single threaded.)

I don't know what the JDK 16 issue is yet when used when using ClassGraph with SpigotMC, but all the unit tests for ClassGraph pass fine on JDK 16, so I suspect something weird is going on with the classloader etc. -- I'll work on that as soon as I get logs.

**Reply by JJBRT** [reply]
Thanks!
With Burningwave Core's criteria-based model you can create complex predicates without limits, but classgraph is faster and leaner. I don't think there can be competition between Classgraph and Burningwave Core because they serve different purposes: the former is an optimized engine and specialized for classpath scanning, the second is a library for building frameworks that offers multiple functionalities.

**Reply by lukehutch** [reply]
It's all good :-)  I'm really not as competitive as my comment made it sound! I was just surprised by the claim!

I agree, your library looks like it has some really rich features. How do you accomplish this?:

> [Burningwave Core is] not affected by “the issue of the same class loaded by different classloaders”

**Comment by JJBRT** [comment]
I am only a collaborator of the project and I do not know very well how it is programmed in depth, in general it manages to perform all these functions thanks to a close collaboration between the static components and the dynamic components that are held together by a container. When the authors write "THE MOST POWERFUL CLASSPATH SCANNER" I believe they are referring to the unlimited flexibility offered by the criteria-based model (try taking a look at the various examples [here on the wiki](https://github.com/burningwave/core/wiki) to understand what I mean (ClassHunter section)): so I will propose to the mantainer to modify this sentence. But Burningwave is a Java building framework and you can write your own dependency injection framework with it: if you are interested take a look to my tutorial project [here](https://github.com/JJBRT/advanced-java-tutorials)
